rust/arrow/src/ipc/convert.rs

```diff
@@ -97,6 +97,12 @@ pub fn fb_to_schema(fb: ipc::Schema) -> Schema {
     let len = c_fields.len();
     for i in 0..len {
         let c_field: ipc::Field = c_fields.get(i);
+        match c_field.type_type() {
```

Comment:
@alamb could you take another look at my attempt to add the unimplemented path for big endian?

I am not happy with placing the check in `fb_to_schema` and would have preferred to put it in `get_data_type` but I found no way to pass on the endianness from the schema.

Reply:
I see the problem -- yes, there since the `endianness` is on the shema object, not the field, since the field is all that is passed around there is no way to know what the details of the schema are.

I personally think this code is fine, if a bit un-indeal and could be cleaned up in the future. My only worry is that it would get lost / broken during such cleanup

What would you think about adding a test that triggers the error? Then we could be sure that any future cleanups will not break the check?

Reply:
@alamb thank you for being so nice about it - I was just too lazy to add a test and should receive full scrutiny ;)

This is partly due to the fact that I am not very familiar with flatbuffers and still do not fully understand how to create the appropriate flatbuffer to test this. As a temporary solution, I have added two tests to `ipc::reader` that uses the BigEndian files in `arrow-ipc-stream/integration/1.0.0-bigendian`. The one for decimal fails, the others work. I hope this is okay for now, until I am able to construct the correct schema message to test this directly in `ipc::convert`.

While adding the big endian test for the other types I noticed that the contents are not equal to the json content. That is why the test does not contain an equality check. Thus, there may be problems with Big Endian for other types as well.